File: 0136-single-number/0136-single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        map<int, int> m;
        
        for (auto elem : nums) {
            if (m.count(elem)) {
                m[elem]++;
            } else {
                m[elem] = 1;
            }
        }
        
        for (auto it = m.begin(); it != m.end(); it++) {
            if (it->second == 1) {
                return it->first;
            }
        }
        return 0;
    }
};

File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();
        for (int elem : nums) {
            if (m.containsKey(elem)) {
                m.put(elem, m.get(elem) + 1);
            } else {
                m.put(elem, 1);
            }
        }
        for (Map.Entry<Integer, Integer> elem : m.entrySet()) {
            if (elem.getValue() == 1) {
                return elem.getKey();
            }
        }
        return 0;
    }
}